File: server/models/Project.js
const { DataTypes } = require('sequelize')
const { sequelize } = require('../config/database')

const Project = sequelize.define('Project', {
  id: {
    type: DataTypes.UUID,
    defaultValue: DataTypes.UUIDV4,
    primaryKey: true,
  },
  title: {
    type: DataTypes.STRING,
    allowNull: false,
  },
  description: {
    type: DataTypes.TEXT,
    allowNull: true,
  },
  type: {
    type: DataTypes.STRING,
    allowNull: false,
  },
  status: {
    type: DataTypes.ENUM('in_progress', 'finished'),
    defaultValue: 'in_progress',
  },
}, {
  tableName: 'projects',
  timestamps: true,
})

module.exports = Project

File: go-service/models/project.go
package models

import (
	"time"

	"github.com/google/uuid"
	"gorm.io/gorm"
)

type Project struct {
	ID          string    `json:"id" gorm:"type:uuid;primaryKey"`
	Title       string    `json:"title" gorm:"type:varchar(255);not null"`
	Description *string   `json:"description" gorm:"type:text"`
	Type        string    `json:"type" gorm:"type:varchar(255);not null"`
	Status      string    `json:"status" gorm:"type:varchar(20);default:'in_progress'"`
	CreatedAt   time.Time `json:"createdAt" gorm:"column:createdAt;autoCreateTime"`
	UpdatedAt   time.Time `json:"updatedAt" gorm:"column:updatedAt;autoUpdateTime"`

	Contents []Content `json:"Contents,omitempty" gorm:"foreignKey:ProjectID;constraint:OnDelete:CASCADE"`
}

func (Project) TableName() string {
	return "projects"
}

func (p *Project) BeforeCreate(tx *gorm.DB) error {
	if p.ID == "" {
		p.ID = uuid.New().String()
	}
	if p.Status == "" {
		p.Status = "in_progress"
	}
	return nil
}
